crates/driver/src/logic/eth/eip712.rs

```diff
@@ -8,16 +8,23 @@
 #[derive(Debug, Clone, Copy)]
 pub struct DomainSeparator(pub [u8; 32]);
 
+#[derive(Debug)]
+pub struct DomainFields {
```

Comment:
Technically, all the fields are optional. For example the `SolverTrampoline` contract used for Gelato submissions only uses 2 of them:
https://github.com/cowprotocol/solver-trampoline/blob/0d65e071bd014d38c6c4af3a57364754568b98e3/contracts/SolverTrampoline.sol#L39-L43

I actually like it as-is for now, and we can change it when we introduce ☝️